src/lib/detection_algorithms/cntkfastrcnn/exporter.js

```diff
@@ -0,0 +1,121 @@
+const async = require('async');
+const fs = require('fs');
+const path = require('path');
+const detectionUtils = require('../detectionUtils.js');
+
+// The Exporter interface - provides a mean to export the tagged frames
+// data in the expected data format of the detection algorithm
+// Constructor parameters:
+//  exportDirPath - path to the directory where the exported file will be placed
+//  classes - list of classes supported by the tagged data
+//  taggedFramesCount - number of positive tagged frames
+//  frameWidth - The width (in pixels) of the image frame
+//  frameHeight - The height (in pixels) of the image frame
+//  testSplit - the percent of tragged frames to reserve for test set defaults to 20%
+function Exporter(exportDirPath, classes, taggedFramesCount, frameWidth, frameHeight, testSplit) {
+    var self = this;
+    self.exportDirPath = exportDirPath;
+    self.posDirPath  = path.join(self.exportDirPath, 'positive');
+    self.negDirPath  = path.join(self.exportDirPath, 'negative');
+    self.testDirPath = path.join(self.exportDirPath, 'testImages');
+    self.classes = classes;
+    self.taggedFramesCount = taggedFramesCount;
+    self.frameWidth = frameWidth;
+    self.frameHeight = frameHeight;
+    self.testSplit = testSplit || 0.2;
+    self.posFrameIndex = null;
+    self.testFrameIndecies = null;
+
+    // Prepare everything for exporting (e.g. create metadata files,
+    // directories, ..)    
+    // Returns: A Promise object that resolves when the operation completes
+    this.init = function init() {
+
+         return new Promise((resolve, reject) => {
+            async.waterfall([
+                detectionUtils.generateTestIndecies.bind(null,self.testSplit, self.taggedFramesCount),
+                function updateIndecies(testIndecies, cb) {
+                    self.posFrameIndex = 0;
```

Comment:
Why are you setting this here (posFrameIndex)? Can probably set it at the beginning of the init, no?